modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/GetValueFunction.java

```diff
@@ -36,6 +37,10 @@ public Object apply(String field) {
 			return StringPool.BLANK;
 		}
 
+		if (!_isFieldVisible(field)) {
+			return StringPool.BLANK;
+		}
```

Comment:
@linolaoj, is this code necessary even with the changes implemented by this [commit](https://github.com/liferay-forms/liferay-portal/pull/38/commits/777870d67f1bef2e20e660fa54c85053278781b6)? If so, can you please give me more details. Thanks.

Reply:
@natocesarrego , I had to add it here because when the user submits the form record, there is one more cycle of rule evaluation and adding it here made the hidden fields be saved as empty values.

Reply:
Thanks for your clarification, @linolaoj. 🙂